surface/CMakeLists.txt

```diff
@@ -151,14 +163,20 @@ if(build)
         "include/pcl/${SUBSYS_NAME}/impl/surfel_smoothing.hpp"
         "include/pcl/${SUBSYS_NAME}/impl/texture_mapping.hpp"
         "include/pcl/${SUBSYS_NAME}/impl/poisson.hpp"
+        "include/pcl/${SUBSYS_NAME}/impl/advancing_front.hpp"
         ${HULL_IMPLS}
         )
 
     set(LIB_NAME "pcl_${SUBSYS_NAME}")
     include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include" ${VTK_INCLUDE_DIRS} "${CMAKE_CURRENT_SOURCE_DIR}")
     link_directories(${VTK_LIBRARY_DIRS})
     PCL_ADD_LIBRARY("${LIB_NAME}" "${SUBSYS_NAME}" ${srcs} ${incs} ${impl_incs} ${VTK_SMOOTHING_INCLUDES} ${POISSON_INCLUDES} ${OPENNURBS_INCLUDES} ${ON_NURBS_INCLUDES})
-    target_link_libraries("${LIB_NAME}" pcl_common pcl_search pcl_kdtree pcl_octree ${VTK_LIBRARIES} ${ON_NURBS_LIBRARIES})
+    if(BUILD_surface_afront_debug)
+      target_link_libraries("${LIB_NAME}" pcl_common pcl_search pcl_kdtree pcl_octree pcl_visualization ${VTK_LIBRARIES} ${ON_NURBS_LIBRARIES})
+    else(BUILD_surface_afront_debug)
+      target_link_libraries("${LIB_NAME}" pcl_common pcl_search pcl_kdtree pcl_octree ${VTK_LIBRARIES} ${ON_NURBS_LIBRARIES})
+    endif(BUILD_surface_afront_debug)
+
```

Comment:
Do you see how things are done for QHull in line 180, that's how you should link with VTK. Move your condition below line 194.